File: main/python/algorithms/selection_sort.py
def selection_sort(array):
    n = len(array)
    for i in range(n):
        min_idx = i
        for j in range(i+1, n):
            if array[min_idx] > array[j]:
                min_idx = j
        temp = array[i]
        array[i] = array[min_idx]
        array[min_idx] = temp

    return array

File: main/c++/algorithms/selection_sort.cpp
#include "sort.h"

void selection_sort(int *array, int len) {
    for (int i = 0; i < len; i++) {
        int min_idx = i;
        for (int j = i+1; j < len; j++) {
            if (array[min_idx] > array[j]) {
                min_idx = j;
            }
        }
        int temp = array[i];
        array[i] = array[min_idx];
        array[min_idx] = temp;
    }
}